src/osd/OSD.cc

```diff
@@ -5964,6 +5980,10 @@ void OSD::sched_scrub()
 	break;
       }
 
+      if (!service.check_scrubs_pending()) {
```

Comment:
Does it make sense to do this together with load_check at the beginning of this function to fail fast?